notebooks/concepts/Specviz from splot concept notebook.ipynb

```diff
@@ -0,0 +1,3636 @@
+    "    Similarly, the functionality of the `u` and `v` keys are also available in the notebook as long as the pixel location under the cursor is available - e.g., to do the shift that corresponds to `u` followed by `z` and `6563` as the rest wavelength:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "obs_wl = specviz.get_spectrum_location()[0]  #<- blocks until the user clicks a line\n",
+    "specviz.current_spectrum.redshift = obs_wl/6563-1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similar operations on `specviz.current_spectrum.spectral_axis` allow `v` or other more complex workflows."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. A plugin could be written to handle workflows like `u` and `v` below and have them set the redshift appropriately.  It is not clear these are common enough to need to be in a plugin as opposed to directing users to the notebook as stated above, however."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Line lists "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "While the redshift is useful contextual information, it is most important when used to identify spectral lines.  For this purpose the user needs to be able to input line lists appropriate for their science case. The below describe how to specify such line lists, but regardless the intent is for the user to then see those lines displayed in `specviz`, redshifted to wherver they should be given the value of the `.redshift`/`.radial_velocity` attribute.\n",
+    "\n",
+    "1. In the notebook line lists should be loadable straightforwardly as astropy tables with spectrally-appropriate units.  The most straightforward option is to consume any table that has 'linename' and 'rest' columns:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "linelist_table = astropy.table.Table()\n",
+    "linelist_table['linename'] = ['Halpha', 'Hbeta']\n",
+    "linelist_table['rest'] = [6563, 4862]*u.angstrom\n",
+    "linelist_table['more information'] = ...\n",
+    "\n",
+    "specviz.load_linelist(linelist_table)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. In the UI, either a plugin or a pane specifically dedicated to this (perhaps associated with the \"redshift\" slider?) should present a few options: a \"galaxies\" linelist, a \"stars\" linelist, and a \"load file...\" option.  The first two would be curated lists (ideally in `specutils` but could be in `jdaviz`), while \"load file...\" would prompt the user for a file, which would then be read as `astropy.table.Table.read('/path/to/file')`.  This ensures the same line list tables can be used in the notebook and the UI setting."
+   ]
```

Comment:
This is definitely needed.  For MAST we may want to only use curated list and disable user-supplied versions.  In that case, we would need a way to selectively turn off all user-prompts across the main application but within each plugin.  This example is a clear case where we might want the plugin enabled but not all of it.

Reply:
Hmm. This pushes me towards the UI having only a limited set of options + "load a custom list", but *not* aiming to expose a line-list-maker as a user UI.  That makes it so we don't have to worry about what you're saying here, @havok2063, as the notebook that underlies the MAST view can just load custom MAST line lists without exposing any of it to the user.

Reply:
To be clear, that's just what the notebook says now, but with the added bit that we are *actively* saying we aren't going to add a line list editor - instead MAST would then just have to hide the "load file" option.

Reply:
Perhaps uploading line lists isn't that bad for MAST but I'm hesitant to enable any sort of file uploading onto the MAST server system, so I prefer the use of pre-curated line lists.  

My larger point about the plugins is that for each plugin that you develop that has some sort of user prompt or hook, you should also build in the ability for MAST to disable that particular portion of it but still load and use the plugin.  The answer can't be "well just turn off that plugin".